src/machine/machine_rp2040_pwm.go

```diff
@@ -0,0 +1,494 @@
+	case 1:
+		pos = rp.PWM_CH0_CSR_B_INV_Pos
+		msk = rp.PWM_CH0_CSR_B_INV_Msk
+	}
+	pwm.slice[sliceNum].csr.ReplaceBits(boolToBit(invert)<<pos, msk, 0)
+}
+
+// Set the current PWM counter compare value for one channel
+//
+// The counter compare register is double-buffered in hardware. This means
+// that, when the PWM is running, a write to the counter compare values does
+// not take effect until the next time the PWM slice wraps (or, in
+// phase-correct mode, the next time the slice reaches 0). If the PWM is not
+// running, the write is latched in immediately.
+// Channel is 0 for A, 1 for B.
+//go:inline
+func (pwm *pwms) setChanLevel(slicenum uint8, channel uint8, level uint16) {
+	var pos uint8
+	var mask uint32
+	switch channel {
+	case 0:
+		pos = rp.PWM_CH0_CC_A_Pos
+		mask = rp.PWM_CH0_CC_A_Msk
+	case 1:
+		pos = rp.PWM_CH0_CC_B_Pos
+		mask = rp.PWM_CH0_CC_B_Msk
+	}
+	pwm.slice[slicenum].cc.ReplaceBits(uint32(level)<<pos, mask, 0)
+}
+
+//go:inline
+func (pwm *pwms) getChanLevel(slicenum uint8, channel uint8) (level uint16) {
+	var pos uint8
+	var mask uint32
+	switch channel {
+	case 0:
+		pos = rp.PWM_CH0_CC_A_Pos
+		mask = rp.PWM_CH0_CC_A_Msk
+	case 1:
+		pos = rp.PWM_CH0_CC_B_Pos
+		mask = rp.PWM_CH0_CC_B_Msk
+	}
+
+	level = uint16((pwm.slice[slicenum].cc.Get() & mask) >> pos)
+	return level
+}
+
+//go:inline
+func (pwm *pwms) getWrap(slicenum uint8) (top uint32) {
+	return (pwm.slice[slicenum].top.Get() & rp.PWM_CH0_TOP_CH0_TOP_Msk) >> rp.PWM_CH0_TOP_CH0_TOP_Pos
+}
+
+//go:inline
+func (pwm *pwms) getPhaseCorrect(slicenum uint8) (phCorrect uint32) {
+	return (pwm.slice[slicenum].csr.Get() & rp.PWM_CH0_CSR_PH_CORRECT_Msk) >> rp.PWM_CH0_CSR_PH_CORRECT_Pos
+}
+
+//go:inline
+func (pwm *pwms) getClockDiv(slicenum uint8) (Int, frac uint32) {
+	div := pwm.slice[slicenum].div.Get()
+	return (div & rp.PWM_CH0_DIV_INT_Msk) >> rp.PWM_CH0_DIV_INT_Pos, (div & rp.PWM_CH0_DIV_FRAC_Msk) >> rp.PWM_CH0_DIV_FRAC_Pos
+}
+
+// div: integer value to reduce counting rate by. Must be greater than or equal to 1.
+//
+// cc: counter compare level. Contains 2 channel levels. The 16 LSBs are Channel A's level (Duty Cycle)
+// and the 16 MSBs are Channel B's level.
+//
+// top: Wrap. Highest number counter will reach before wrapping over. usually 0xffff.
+//
+// csr: Clock mode. PWM_CH0_CSR_DIVMODE_xxx registers have 4 possible modes, of which Free-running is used.
+// csr contains output polarity bit at PWM_CH0_CSR_x_INV where x is the channel.
+// csr contains phase correction bit at PWM_CH0_CSR_PH_CORRECT_Msk.
+// csr contains PWM enable bit at PWM_CH0_CSR_EN. If not enabled PWM will not be active.
+//
+// ctr: PWM counter value.
+type pwmGroup []struct {
+	csr, cc, ctr, div, top *volatile.Register32
+}
+
+//go:inline
+func (pwm *pwms) initSlice() {
+	pwm.slice = make(pwmGroup, 8)
+	s := pwm.slice
+	hw := pwm.hw
+	// PWM0
+	s[0].cc = &hw.CH0_CC
+	s[0].csr = &hw.CH0_CSR
+	s[0].ctr = &hw.CH0_CTR
+	s[0].div = &hw.CH0_DIV
+	s[0].top = &hw.CH0_TOP
```

Comment:
Hmm, it's a bit ugly that the RP2040 SVD file doesn't contain an array of slices.
I think this would be an appropriate use of `unsafe`:

```go
type pwmGroup struct {
    CSR volatile.Register32
    DIV volatile.Register32
    CTR volatile.Register32
    CC  volatile.Register32
    TOP volatile.Register32
}

func getPWMGroup(index uintptr) *pwmGroup {
    return (*pwmGroup)(unsafe.Pointer(rp.PWM))
}
```

You can then use it like `getPWMGroup(0).CSR.Set(...)`.